java-security-test/src/main/java/com/sap/cloud/security/test/JwtGenerator.java

```diff
@@ -220,6 +220,28 @@ public JwtGenerator withClaimValue(String claimName, JsonObject object) {
 		return this;
 	}
 
+	/**
+	 * Sets the claims with the given names to the given string value. Note: for
+	 * overwriting client Id claim, "azp" claim value should be overwritten instead
+	 * of deprecated "cid"
+	 *
+	 * @param claimName
+	 *            the name of the claim to be set.
+	 * @param map
+	 *            map of key value pairs of claims to be set.
+	 * @return the builder object.
+	 * @throws JsonParsingException
+	 *             if the given object does not contain valid json.
+	 */
+	public JwtGenerator withClaimValue(String claimName, Map<String, String> map) {
```

Comment:
Thats correct and good to have, but is this used somewhere, e.g. in IasTokenAuthenticatorX509Test.java

Reply:
there is no other easy way provided in the JwtGenerator to create nested object. I've used it to generate the token with x5t claim for testing resources.